### DISCUSSION TITLE
Is SemanticToken support a possability?

### DISCUSSION THREAD

**Discussion by Ryex** [mkslanc/ace-linters#104] [Q&A] [answered]
I'm wondering if [Semantic Token](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens) support is on the roadmap?

I'm currently working of a project to create an editor and emulator for an toy MIPS variant used as a scripting language in a game. that variant has a working LSP built by the community that I managed to get working via wasm in a webworker. The LSP has beautiful  Semantic Token support but doesn't support highlighting. As such I've had to build my own highlighting rules. These rules can't capture the same context as the lsp though.

**Comment by mkslanc** [comment, marked as answer]
It wasn't in roadmap, but seems pretty interesting task, so I will resolve it ASAP (when I would have time considering other projects)

**Comment by mkslanc** [comment]
Last (1.2.0) version should allow to use semantic tokens. It's a bit experimental, so by default turned off, but you could use it by setting in on like `LanguageProvider.create(worker, {functionality: {semanticTokens: true}})`. Feel free to drop new issues, if there are any 👍
